assembl/scripts/ini_files.py

```diff
@@ -144,72 +166,392 @@ def main():
+            for option, value2 in second.items(section):
+                if not first.has_option(section, option):
+                    if not existing_only:
+                        ensureSection(diff, section)
+                        diff.set(section, option, value2)
+                    continue
+                value1 = first.get(section, option)
+                if value1 != value2 and '%(' in value1:
+                    # try to interpolate, and see if it would amount to the same.
+                    try:
+                        value1 = interpolating._interpolate(section, option, value1, vars)
+                    except Exception as e:
+                        pass
+                if value1 != value2:
+                    ensureSection(diff, section)
+                    diff.set(section, option, value2)
+    return diff
+
+
+def compose(rc_filename, random_file=None):
+    """Compose local.ini from the given .rc file"""
+    rc_info = combine_rc(rc_filename)
+    ini_sequence = rc_info.get('ini_files', None)
+    assert ini_sequence, "Define ini_files"
+    ini_sequence = ini_sequence.split()
+    base = Parser()
+    random_file = random_file or rc_info.get('random_file', RANDOM_FILE)
+    for overlay in ini_sequence:
+        if overlay == 'RC_DATA':
+            overlay = rc_to_ini(rc_info)
+        elif overlay.startswith('RANDOM'):
+            templates = overlay.split(':')[1:]
+            overlay = populate_random(
+                random_file, templates, extract_saml_info(rc_info))
+        combine_ini(base, overlay)
+    return base
+
+
+def extract_random_templates(ini_files):
+    ini_files = ini_files or ''
+    random = [f for f in ini_files.split() if f.startswith("RANDOM")]
+    if random:
+        assert len(random) == 1, "Only a single random element in ini_files"
+        return random[0].split(":")[1:]
+    return []
+
+
+def migrate(rc_filename, expected_ini, random_file=None, target_dir=None):
+    """Create a overlay.rc file from the local.ini and a base .rc file"""
+    expected_ini = asParser(expected_ini)
+    rc_data = combine_rc(rc_filename)
+    ini_files = rc_data.get('ini_files', None)
+    random_file = random_file or rc_data.get('random_file', RANDOM_FILE)
+    templates = extract_random_templates(ini_files)
+    if templates:
+        random_data = populate_random(
+            random_file, templates, extract_saml_info(rc_data))
+        migrate_beaker_config(random_data, expected_ini)
+        random_data = combine_ini(random_data, expected_ini, False)
+        with open(random_file, 'w') as f:
+            random_data.write(f)
+    else:
+        random_file = None
+    base = compose(rc_filename, random_file)
+    diff = diff_ini(base, expected_ini)
+    return iniconfig_to_rc(diff, extends=rc_filename, target_dir=target_dir)
+
+
+def main():
+    parser = ArgumentParser(prog='ini_files.py')
```

Comment:
good idea. I was guarding against the default of prog[0], which could be assembl_ini_files